lib/ansible/modules/find.py

```diff
@@ -56,6 +56,8 @@
     contains:
         description:
             - A regular expression or pattern which should be matched against the file content.
+            - Depending on I(read_whole_file), the regular expression or pattern MUST either match the
+              ENTIRE file content or an ENTIRE line, not just parts of it.
```

Comment:
Hmm, this sounds wrong to me. When `read_whole_file=true`, `re.search()` is used, which searches for the regex *somewhere* in the whole file. When `read_whole_file=false`, `re.match()` is used for every line, which searches for the regex at the start of each line.

So the pattern must neither match the entire file when `read_whole_file=true`, nor an entire line (but the beginning of a line) when `read_whole_file=false`.

Reply:
Oh … I see your point, I messed up while writing this.

For `read_whole_file=true`, I thought that this means "match against whole file", which is, when reading the code, obviously not correct.

For `read_whole_file=false`, I neither read that `re.match` is used according to its description, nor I thought that it does just match from the start, but that explains the "problem" I had.

Thanks for you review, I will change the PR accordingly.